File: Algorithms/Sorting/Insertion Sort - Part 1/my_solution.py
def PrintArray(arr):
    """Printing each array elements."""
    for i in range(len(arr)):
        print(str(arr[i]) + ' ' if i != len(arr)-1 else str(arr[i]), end='')


def Sort(arr):
    """Sorting array by ascending."""
    for i in reversed(range(len(arr))):
        key = arr[i]
        j = i-1
        
        while (j >= 0 and key <= arr[j]):
            arr[j+1] = arr[j]
            j -= 1
            PrintArray(arr)
            print('')

        arr[j+1] = key

    PrintArray(arr)


n = int(input())
numbers = list(map(int, input().split()))
Sort(numbers)

File: Algorithms/Sorting/Insertion Sort - Part 1/my_solution.cpp
#include <iostream>
#include <vector>
using namespace std;

void PrintArray(vector<int> &arr);
void Sort(vector<int> &arr);

int main()
{
    int n = 0;
    cin >> n;
    vector<int> numbers(n, 0);

    for (int i = 0; i < n; i++) {
        cin >> numbers[i];
    }

    Sort(numbers);
}

void PrintArray(vector<int> &arr) {
    for (int i = 0; i < arr.size(); i++) {
        cout << (i == arr.size()-1 ? to_string(arr[i]) : to_string(arr[i]) + " ");
    }
}

void Sort(vector<int> &arr) {
    for (int i = arr.size()-1; i >= 0; i--) {
        int key = arr[i];
        int j = i-1;

        while (j >= 0 && key <= arr[j]) {
            arr[j+1] = arr[j];
            j--;
            PrintArray(arr);
            cout << endl;
        }
        arr[j+1] = key;
    }

    PrintArray(arr);
}